src/Table/TableCell.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext } from 'react'
+import styled from 'styled-components'
+import css from '@styled-system/css'
+import PropTypes from 'prop-types'
+
+import { Box } from '@ivoryio/kogaio'
+import styles from './TableCell.style'
+import { TableContext } from './Table'
+
+const HeaderCellStyle = styled(Box)(
+  css(styles.header),
+  ({ gridTemplate, ...props }) => ({
+    gridAutoColumns: gridTemplate,
+    ...props
```

Comment:
I'm not sure that's what you mean. `styled` will pass the props to the underlying component (`Box` in this case). Adding them here will try to add them as style.